ADR/0028-component-dependencies.md

```diff
@@ -0,0 +1,179 @@
+* As with Components in general, the "depends on" references are Application-scoped. Declared relationships between Components in different Applications are not to be considered valid. Declared relationships between Components in different namespaces are not to be considered valid.
+
+**PR Groups**: Introduce a new convention for PRs that lets one PR declare that it is related to another PR.
+
+* Treat the **source branch name** of the PR as a `@`-delimited string, and use the first element of that
+  list as the PR Group name.
+* A new set of functionality for [integration-service] uses PR Groups to guide testing of related
+  PRs.
+* The set of PRs in a PR Group form a flat set.
+* PR Groups are scoped to Component repos under a single Application. Even if PRs across different
+  applications share the same **source branch name**, they shouldn't be considered part of the
+  same PR Group.
+* For example, if the source branch name for a PR is `my-feature@change-1`, then the PR Group
+  name should be interpreted as `my-feature`.
+* A source branch name with no `@` is the degenerate case. For example, if the source branch
+  name for a PR is `my-feature`, then the PR Group name should be interpreted as `my-feature`.
+* Since all PRs have a source branch name, all PRs are trivially members of a PR Group.
+
+Important to understand: the declared dependencies as defined on the Components' `depends on` lists and PR Groups are two different things:
+
+* Component dependencies are declarations about which Components' builds need each other. Automation uses this to create PRs in PR Groups.
+  * You say in english: “Builds of Component A depend on builds of Component B.”
+* PR Groups can be supplied by the user without any declared Component dependencies:
+  * You say in english: “PR #1 and PR #2 are in the same PR Group.”
+
+**CEL expressions on Build Pipelines**: [Pipelines as Code](https://pipelinesascode.com/)(PaC) supports [advanced event matching](https://pipelinesascode.com/docs/guide/authoringprs/#advanced-event-matching) by way of a `pipelinesascode.tekton.dev/on-cel-expression` annotation on the PipelineRun. When used with the `.pathChanged` field, PaC can be directed to only create a PipelineRun for events which include changes to the declared context directory in the git repository. This enables us to only run PipelineRuns that are relevant to Components that change in a given push or pull request event. By default, this is ".": the entire git repository.
+
+* [build-service] should start setting `pipelinesascode.tekton.dev/on-cel-expression: "*".pathChanged()`, which should match today's default behavior for running all pipelineruns on changes to any file in the repo and our user documentation should be updated to instruct users to modify that path for components that should only build on changes to certain files to avoid unnecessary rebuilds.
+
+See also [RHTAP-371](https://issues.redhat.com/browse/RHTAP-371).
```

Comment:
There is very similar (if not duplicate) [issue](https://issues.redhat.com/browse/STONEBLD-1720). However, there is a planned discussion on architecture call regarding expediency of using `on-cel-expression` annotations for multi component git repositories.